src/main.py

```diff
@@ -156,3 +172,37 @@ def edit_subscription():
         db.serialize(user)
         return redirect(url_for('index'))
     return render_template("edit_subscription.html", form=form, the_title="Edit Subscription - Paint Drying")
+
+
+@app.route("/admin_panel", methods=['GET', 'POST'])
+def admin_panel():
+    if request.method == 'GET':
+        if request.args.get('suggest-services') == 'suggest':
+            clients = db.basic_db.get_clients()
+            print(clients)
+            for client in clients:
+                mail_text = get_propose_mail_text(client['id'], Database(db))
+                send_mail(client['email'],
+                          'Suggestion',
+                          mail_text.__str__(),
+                          [])
+            return render_template("admin_panel.html", the_title="Paint Drying", notification="Mails sent")
+        if request.args.get('send-invoice') == 'send':
+            clients = db.basic_db.get_clients()
+            if len(clients) > 0:
+                client = db.get_client_by_email(clients[0]['email'])
+                file_name = 'invoice.pdf'
+                invoice = Invoice(client)
+                invoice.save_pdf('invoice.pdf')
```

Comment:
In line 194 you define a variable called "`file_name`", but here the filename is given as a string